### PR TITLE
Change feature of Keepers

### PR DESCRIPTION
**Pull request by VadimDyachenko**
connect #99

### CODE REVIEW

File: src/main/java/juja/microservices/gamification/entity/KeeperDTO.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * @author Vadim Dyachenko
+ */
+@Getter
+@EqualsAndHashCode
+public class KeeperDTO {
+    @NotEmpty
+    private String uuid;
+    @NotEmpty
+    private List<String> direction;
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/entity/KeeperDTO.java]: please use plural for the name of list - directions.

File: src/main/java/juja/microservices/gamification/service/AchievementService.java [hunk 2]
```diff
@@ -185,25 +188,44 @@ private void checkUsers(CodenjoyRequest request) {
         return result;
     }
 
+    @Scheduled(cron = "${cron.expression}")
+    private void addScheduledThanksKeeper() {
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/service/AchievementService.java:192]: why this method private? where is it called from?

Comment (OlgaKulikova) [reply src/main/java/juja/microservices/gamification/service/AchievementService.java:192]: Do you mean that scheduler can call it if it's private?

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -5,6 +5,7 @@ spring.data.mongodb.password=root
 spring.data.mongodb.database=gamification
 spring.data.mongodb.repositories.enabled=true
 spring.data.mongodb.uri=mongodb://localhost:27017/gamification
-user.baseURL=http://localhost:8090
+keepers.baseURL=http://localhost:8090
 endpoint.keepers=/keepers
```
Comment (OlgaKulikova) [inline src/main/resources/application.properties]: please consider that all services will have api version

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -5,6 +5,7 @@ spring.data.mongodb.password=root
...
-spring.output.ansi.enabled=ALWAYS
\ No newline at end of file
+spring.output.ansi.enabled=ALWAYS
+cron.expression=* 0 0 * * SUN
```
Comment (OlgaKulikova) [inline src/main/resources/application.properties]: please add a comment next to cron - how often it works

**Review by OlgaKulikova — CHANGES_REQUESTED** [review]

**Comment by OlgaKulikova** [comment]
@VadimDyachenko please do update of upstream.

File: src/main/java/juja/microservices/gamification/entity/KeeperDTO.java (hunk 1 above)
Comment (VadimDyachenko) [reply src/main/java/juja/microservices/gamification/entity/KeeperDTO.java]: done

File: src/main/java/juja/microservices/gamification/service/AchievementService.java (hunk 2 above)
Comment (VadimDyachenko) [reply src/main/java/juja/microservices/gamification/service/AchievementService.java:192]: I checked it. This is work well

File: src/main/resources/application.properties (hunk 3 above)
Comment (VadimDyachenko) [reply src/main/resources/application.properties]: done

File: src/main/resources/application.properties (hunk 4 above)
Comment (VadimDyachenko) [reply src/main/resources/application.properties]: done

**Comment by VadimDyachenko** [comment]
@OlgaKulikova Done!

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -6,6 +6,7 @@ spring.data.mongodb.database=gamification
 spring.data.mongodb.repositories.enabled=true
 spring.data.mongodb.uri=mongodb://localhost:27017/gamification
 keepers.baseURL=http://localhost:8090
-endpoint.keepers=/keepers
+endpoint.keepers=/v1/keepers
 spring.output.ansi.enabled=ALWAYS
-cron.expression=* 0 0 * * SUN
\ No newline at end of file
+cron.expression=* 0 0 * * SUN
+#Schedueled task will run every Sunday at 00-00
```
Comment (OlgaKulikova) [inline src/main/resources/application.properties]: Comments always should be written above code they describe. Put it up to one string.

**Review by OlgaKulikova — APPROVED** [review]

**Pull request merged**